internal/js/modules/k6/browser/common/network_manager.go

```diff
@@ -872,24 +874,60 @@ func (m *NetworkManager) ContinueRequest(requestID fetch.RequestID) error {
 	return nil
 }
 
-func (m *NetworkManager) FulfillRequest(requestID fetch.RequestID, params fetch.FulfillRequestParams) {
-	action := fetch.FulfillRequest(requestID, params.ResponseCode).
-		WithResponseHeaders(params.ResponseHeaders).
-		WithResponsePhrase(params.ResponsePhrase).
-		WithBody(params.Body)
+func (m *NetworkManager) FulfillRequest(request *Request, opts *FulfillOptions) error {
+	responseCode := int64(http.StatusOK)
+	if opts != nil && opts.Status != 0 {
+		responseCode = opts.Status
+	}
+
+	action := fetch.FulfillRequest(request.interceptionID, responseCode)
+
+	if opts.ContentType != "" {
```

Comment:
Still need to check for `nil` `opts`? Maybe we need to check for `nil` opts at the beginning and `fulfill` with no extra options? Otherwise adding the options fields that aren't `nil`.

Reply:
Updated in https://github.com/grafana/k6/pull/4961/commits/4452b9e7a6bd7196a429689bd91ca7365035d56e